Zend/zend_closures.c

```diff
@@ -474,10 +485,15 @@ static HashTable *zend_closure_get_gc(zval *obj, zval **table, int *n) /* {{{ */
 {
 	zend_closure *closure = (zend_closure *)Z_OBJ_P(obj);
 
-	*table = Z_TYPE(closure->this_ptr) != IS_NULL ? &closure->this_ptr : NULL;
-	*n = Z_TYPE(closure->this_ptr) != IS_NULL ? 1 : 0;
-	return (closure->func.type == ZEND_USER_FUNCTION) ?
-		closure->func.op_array.static_variables : NULL;
+	if (closure->func.type == ZEND_USER_FUNCTION) {
+		ZVAL_COPY_VALUE(&closure->func.op_array.static_variables[-1], &closure->this_ptr);
```

Comment:
The code in many places below would be simpler if `static_variables[last_static_var]` was used for this instead (so that static_variables stays the start of the allocation).